controlplane/kubeadm/internal/etcd/etcd.go

```diff
@@ -53,6 +54,7 @@ type Client struct {
 	EtcdClient etcd
 	Endpoint   string
 	LeaderID   uint64
+	KVClient   clientv3.KV
```

Comment:
I think we can avoid having `KVClient` as a field by calling `clientv3.NewKV(etcdClient)` inside `HealthCheck`

Reply:
I cannot do this because EtcdClient is not clientV3.client. Actually, I used KV interface to have a way to access clientv3.Client and to also have a fake one.

Reply:
Ok, but the call `newEtcdClient(ctx, etcdClient, etcdClient)` shows that `etcdClient` implements `KVClient`, and passing the same argument twice to a func is really strange.

Why not use a cast inside of `HealthCheck` instead of `NewKV`?
Also, for the fake, why not to add this implementation to a fake etcdClient?

Reply:
You are right, that worked thanks